connect-inject/endpoints_controller_test.go

```diff
@@ -849,6 +851,525 @@ func TestReconcileDeleteEndpoint(t *testing.T) {
 	}
 }
 
+func TestFilterAgentPods(t *testing.T) {
```

Comment:
we should add `t.Parallel()` to all tests I think.

Reply:
DONE!